perun-core/src/main/java/cz/metacentrum/perun/core/api/UsersManager.java

```diff
@@ -463,6 +463,51 @@
 	 */
 	List<UserExtSource> getUserExtSourcesByIds(PerunSession sess, List<Integer> ids) throws PrivilegeException;
 
+	/**
+	 * Return userExtSource for specific attribute definition (specified by id) and unique value.
+	 * If not found, throw and exception.
+	 *
+	 * It looks for exactly one value of the specific attribute type:
+	 * - Integer -> exactly match
+	 * - String -> exactly match
+	 * - Map -> exactly match of "key=value"
+	 * - ArrayList -> exactly match of one of the value
+	 *
+	 * @param sess Perun session
+	 * @param attrId attribute id used for founding attribute definition which has to exists, be unique and in userExtSource namespace
+	 * @param uniqueValue value used for searching
+	 *
+	 * @return userExtSource found by attribute id and it's unique value
+	 *
+	 * @throws UserExtSourceNotExistsException if userExtSource can't be found
+	 * @throws AttributeNotExistsException if attribute can't be found by it's id
+	 * @throws PrivilegeException if the principal does not have sufficient rights to call this method
+	 */
+	UserExtSource getUserExtSourceByUniqueAttributeValue(PerunSession sess, int attrId, String uniqueValue) throws AttributeNotExistsException, UserExtSourceNotExistsException, PrivilegeException;
```

Comment:
I think it would be better to return null if no matching userExtSource is found. It is better for API clients to handle such situations, but it's your call.

Reply:
Not changed